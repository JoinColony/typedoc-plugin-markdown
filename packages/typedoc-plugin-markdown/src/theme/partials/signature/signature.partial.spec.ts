import {
  DeclarationReflection,
  ProjectReflection,
  SignatureReflection,
} from 'typedoc';
import { signaturePartial } from './signature.partial';

describe(`Signature Partial:`, () => {
  let project: ProjectReflection;

  beforeAll(async () => {
    project = global.getProject('signatures.ts');
  });

  test(`should compile signature with a flag`, () => {
    expect(
      signaturePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'privateFunction',
          ) as DeclarationReflection
        ).signatures[0],
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with params'`, () => {
    expect(
      signaturePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'functionWithParameters',
          ) as DeclarationReflection
        ).signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile function that returns an object'`, () => {
    expect(
      signaturePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'functionReturningAnObject',
          ) as DeclarationReflection
        ).signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile a promise that returns an object'`, () => {
    expect(
      signaturePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'promiseReturningAnObject',
          ) as DeclarationReflection
        ).signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile a promise that returns a symbol'`, () => {
    expect(
      signaturePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'promiseReturningASymbol',
          ) as DeclarationReflection
        ).signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile function that returns a function'`, () => {
    expect(
      signaturePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'functionReturningAFunction',
          ) as DeclarationReflection
        ).signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with rest params'`, () => {
    expect(
      signaturePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'functionWithRest',
          ) as DeclarationReflection
        ).signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with optional params'`, () => {
    expect(
      signaturePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'functionWithOptionalParam',
          ) as DeclarationReflection
        ).signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with union types'`, () => {
    expect(
      signaturePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'functionWithUnionTypes',
          ) as DeclarationReflection
        ).signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with default values'`, () => {
    expect(
      signaturePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'functionWithDefaults',
          ) as DeclarationReflection
        ).signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with @return comments'`, () => {
    expect(
      signaturePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'commentsInReturn',
          ) as DeclarationReflection
        ).signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile named parameters'`, () => {
    expect(
      signaturePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'functionWithNamedParams',
          ) as DeclarationReflection
        ).signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile named parameters with comments'`, () => {
    expect(
      signaturePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'functionWithNamedParamsAndComments',
          ) as DeclarationReflection
        ).signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile pipes in params and comments'`, () => {
    expect(
      signaturePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'functionWithPipesInParamsAndComments',
          ) as DeclarationReflection
        ).signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile function with reference type'`, () => {
    expect(
      signaturePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'functionWithReferenceType',
          ) as DeclarationReflection
        ).signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile function with nested typen params'`, () => {
    expect(
      signaturePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'functionWithNestedParams',
          ) as DeclarationReflection
        ).signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile class with constructor'`, () => {
    expect(
      signaturePartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'ClassWithConstructor',
          ) as DeclarationReflection
        ).children[0].signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });
});
