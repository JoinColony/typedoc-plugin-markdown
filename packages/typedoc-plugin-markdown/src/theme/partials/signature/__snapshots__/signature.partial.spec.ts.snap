// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signature Partial: should compile a promise that returns a symbol' 1`] = `
"**promiseReturningASymbol**(): [\`Promise\`]( Promise )<[\`User\`]( User )\\\\>

#### Returns

[\`Promise\`]( Promise )<[\`User\`]( User )\\\\>

{ sourcesPartial }"
`;

exports[`Signature Partial: should compile a promise that returns an object' 1`] = `
"**promiseReturningAnObject**(): [\`Promise\`]( Promise )<{ \`data\`: \`string\` ; \`id\`: \`string\`  }\\\\>

#### Returns

[\`Promise\`]( Promise )<{ \`data\`: \`string\` ; \`id\`: \`string\`  }\\\\>

{ sourcesPartial }"
`;

exports[`Signature Partial: should compile class with constructor' 1`] = `
"**new ClassWithConstructor**(\`x\`: \`string\`, \`y\`: \`string\`)

#### Parameters

{ parameterTablePartial }

{ sourcesPartial }"
`;

exports[`Signature Partial: should compile function that returns a function' 1`] = `
"**functionReturningAFunction**(): <T\\\\>(\`x\`: \`string\`) => \`boolean\`

{ commentPartial }

#### Returns

\`fn\`

Return comments


{ signaturePartial }

{ sourcesPartial }"
`;

exports[`Signature Partial: should compile function that returns an object' 1`] = `
"**functionReturningAnObject**(): \`Object\`

{ commentPartial }

#### Returns

\`Object\`

Return comments


{ propertyTablePartial }

{ sourcesPartial }"
`;

exports[`Signature Partial: should compile function with nested typen params' 1`] = `
"**functionWithNestedParams**(\`params\`: \`Object\`, \`context\`: \`any\`): \`boolean\`

{ commentPartial }

#### Parameters

{ parameterTablePartial }

#### Returns

\`boolean\`

{ sourcesPartial }"
`;

exports[`Signature Partial: should compile function with reference type' 1`] = `
"**functionWithReferenceType**(\`descriptor\`: [\`TypedPropertyDescriptor\`]( TypedPropertyDescriptor )<\`any\`\\\\>): \`boolean\`

#### Parameters

{ parameterTablePartial }

#### Returns

\`boolean\`

{ sourcesPartial }"
`;

exports[`Signature Partial: should compile named parameters with comments' 1`] = `
"**functionWithNamedParamsAndComments**(\`__namedParameters?\`: \`Object\`, \`anotherParam\`: \`string\`): \`void\`

{ commentPartial }

#### Parameters

{ parameterTablePartial }

#### Returns

\`void\`

{ sourcesPartial }"
`;

exports[`Signature Partial: should compile named parameters' 1`] = `
"**functionWithNamedParams**(\`__namedParameters\`: \`Object\`): \`string\`

#### Parameters

{ parameterTablePartial }

#### Returns

\`string\`

{ sourcesPartial }"
`;

exports[`Signature Partial: should compile pipes in params and comments' 1`] = `
"**functionWithPipesInParamsAndComments**(\`n\`: \`number\`): \`number\` \\\\| \`\`null\`\`

{ commentPartial }

#### Parameters

{ parameterTablePartial }

#### Returns

\`number\` \\\\| \`\`null\`\`

{ sourcesPartial }"
`;

exports[`Signature Partial: should compile signature with @return comments' 1`] = `
"**commentsInReturn**(): \`boolean\`

{ commentPartial }

#### Returns

\`boolean\`

Return comments


{ sourcesPartial }"
`;

exports[`Signature Partial: should compile signature with a flag 1`] = `
"\`Private\` **privateFunction**(): \`string\`

{ commentPartial }

#### Returns

\`string\`

{ sourcesPartial }"
`;

exports[`Signature Partial: should compile signature with default values' 1`] = `
"**functionWithDefaults**(\`valueA?\`: \`string\`, \`valueB?\`: \`number\`, \`valueC?\`: \`number\`, \`valueD?\`: \`boolean\`, \`valueE?\`: \`boolean\`, \`valueF?\`: \`string\`): \`string\`

{ commentPartial }

#### Parameters

{ parameterTablePartial }

#### Returns

\`string\`

{ sourcesPartial }"
`;

exports[`Signature Partial: should compile signature with optional params' 1`] = `
"**functionWithOptionalParam**(\`requiredParam\`: \`string\`, \`optionalParam?\`: \`string\`): \`void\`

{ commentPartial }

#### Parameters

{ parameterTablePartial }

#### Returns

\`void\`

{ sourcesPartial }"
`;

exports[`Signature Partial: should compile signature with params' 1`] = `
"**functionWithParameters**(\`paramZ\`: \`string\`, \`paramG\`: \`any\`, \`paramA\`: \`any\`): \`number\`

{ commentPartial }

#### Parameters

{ parameterTablePartial }

#### Returns

\`number\`

{ sourcesPartial }"
`;

exports[`Signature Partial: should compile signature with rest params' 1`] = `
"**functionWithRest**(...\`rest\`: \`string\`[]): \`string\`

{ commentPartial }

#### Parameters

{ parameterTablePartial }

#### Returns

\`string\`

{ sourcesPartial }"
`;

exports[`Signature Partial: should compile signature with union types' 1`] = `
"**functionWithUnionTypes**(\`arg\`: \`number\` \\\\| \`boolean\`[], ...\`args\`: (\`string\` \\\\| \`number\`)[]): \`any\`

#### Parameters

{ parameterTablePartial }

#### Returns

\`any\`

{ sourcesPartial }"
`;
